### PR TITLE
`hack-vendor.sh` improvements

### PR DESCRIPTION
**Pull request by ellismg**
- Add a flag to keep the changes to the Gopkg.toml file.
- Keep constraints.

### CODE REVIEW

File: hack-vendor/cmd/change-constraint/main.go [hunk 1]
```diff
@@ -63,6 +63,13 @@ func main() {
 		os.Exit(1)
 	}
 
+	for idx, constraint := range gopkg.Constraint {
+		if constraint.Name == name {
+			gopkg.Constraint = append(gopkg.Constraint[:idx], gopkg.Constraint[idx+1:]...)
```
Comment (pgavlin) [inline hack-vendor/cmd/change-constraint/main.go]: A small comment here would be nice--easy to miss what's going on.

File: hack-vendor/cmd/change-constraint/main.go [hunk 2]
```diff
@@ -106,12 +113,13 @@ func main() {
 }
 
 type Gopkg struct {
```
Comment (pgavlin) [inline hack-vendor/cmd/change-constraint/main.go]: Any other elements we need to be aware of?

**Comment by stack72** [comment]
@ellismg is this needed anymore now that we have moved to go modules?

**Comment by ellismg** [comment]
> @ellismg is this needed anymore now that we have moved to go modules?

Thankfully, no!

**Comment by lukehoban** [comment]
>  @ellismg reopened this 36 minutes ago

Intentional?

**Comment by ellismg** [comment]
> Intentional?

Nope

**Pull request closed without merging**